Example/Kommunicate/AppDelegate.swift

```diff
@@ -116,21 +87,31 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
     }
 
     func registerForNotification() {
-        if #available(iOS 10.0, *) {
-            UNUserNotificationCenter.current().requestAuthorization(options:[.badge, .alert, .sound]) { (granted, error) in
+        UNUserNotificationCenter.current().requestAuthorization(options:[.badge, .alert, .sound]) { (granted, error) in
 
-                if granted {
-                    DispatchQueue.main.async {
-                        UIApplication.shared.registerForRemoteNotifications()
-                    }
+            if granted {
+                DispatchQueue.main.async {
+                    UIApplication.shared.registerForRemoteNotifications()
                 }
             }
-        } else {
-            // Fallback on earlier versions
-            let settings = UIUserNotificationSettings(types: [.alert, .badge, .sound], categories: nil)
-            UIApplication.shared.registerUserNotificationSettings(settings)
-            UIApplication.shared.registerForRemoteNotifications()
+        }
+    }
 
+    func userNotificationCenter(_ center: UNUserNotificationCenter, willPresent notification: UNNotification, withCompletionHandler completionHandler: @escaping (UNNotificationPresentationOptions) -> Void) {
+        let service = KMPushNotificationService()
+        guard !service.isKommunicateNotification(notification.request.content.userInfo) else {
+            return
```

Comment:
Here we'll have to call `service.notificationArrived()` or `service.processPushNotification()` otherwise the notification alert won't be shown when MQTT is disconnected. And after that, call `completionHandler` with empty array.
Tested after these changes it is working fine.